tests/test_coreg.py

```diff
@@ -126,3 +132,235 @@ def test_only_paths():
+
+        aligned_dem = icp.apply(self.tba.data, self.ref.transform)
+
+        assert aligned_dem.shape == self.ref.data.squeeze().shape
+
+    def test_pipeline(self):
+        warnings.simplefilter("error")
+
+        # Create a pipeline from two coreg methods.
+        pipeline = coreg.CoregPipeline([coreg.BiasCorr(), coreg.ICP(max_iterations=3)])
+        pipeline.fit(**self.fit_params)
+
+        aligned_dem = pipeline.apply(self.tba.data, self.ref.transform)
+
+        assert aligned_dem.shape == self.ref.data.squeeze().shape
+
+        # Make a new pipeline with two bias correction approaches.
+        pipeline2 = coreg.CoregPipeline([coreg.BiasCorr(), coreg.BiasCorr()])
+        # Set both "estimated" biases to be 1
+        pipeline2.pipeline[0]._meta["bias"] = 1
+        pipeline2.pipeline[1]._meta["bias"] = 1
+
+        # Assert that the combined bias is 2
+        pipeline2.to_matrix()[2, 3] == 2.0
+
+    def test_coreg_add(self):
+        warnings.simplefilter("error")
+        # Test with a bias of 4
+        bias = 4
+
+        bias1 = coreg.BiasCorr()
+        bias2 = coreg.BiasCorr()
+
+        # Set the bias attribute
+        for bias_corr in (bias1, bias2):
+            bias_corr._meta["bias"] = bias
+
+        # Add the two coregs and check that the resulting bias is 2* bias
+        bias3 = bias1 + bias2
+        assert bias3.to_matrix()[2, 3] == bias * 2
+
+        # Make sure the correct exception is raised on incorrect additions
+        try:
+            bias1 + 1
+        except ValueError as exception:
+            if "Incompatible add type" not in str(exception):
+                raise exception
+
+        # Try to add a Coreg step to an already existing CoregPipeline
+        bias4 = bias3 + bias1
+        assert bias4.to_matrix()[2, 3] == bias * 3
+
+        # Try to add two CoregPipelines
+        bias5 = bias3 + bias3
+        assert bias5.to_matrix()[2, 3] == bias * 4
+
+    def test_subsample(self):
+        warnings.simplefilter("error")
+
+        # Test subsampled bias correction
+        bias_sub = coreg.BiasCorr()
+
+        # Fit the bias using 50% of the unmasked data using a fraction
+        bias_sub.fit(**self.fit_params, subsample=0.5)
+        # Do the same but specify the pixel count instead.
+        # They are not perfectly equal (np.count_nonzero(self.mask) // 2 would be exact)
+        # But this would just repeat the subsample code, so that makes little sense to test.
+        bias_sub.fit(**self.fit_params, subsample=self.tba.data.size // 2)
+
+        # Do full bias corr to compare
+        bias_full = coreg.BiasCorr()
+        bias_full.fit(**self.fit_params)
+
+        # Check that the estimated biases are similar
+        assert abs(bias_sub._meta["bias"] - bias_full._meta["bias"]) < 0.1
+
+        # Test ICP with subsampling
+        icp_full = coreg.ICP(max_iterations=20)
+        icp_sub = coreg.ICP(max_iterations=20)
```

Comment:
missing a subsample here?

Reply:
Subsampling is done in the `icp_sub.fit()` step. It's only the `.fit()` method that uses subsampling.